c/tskit/tables.c

```diff
@@ -6769,19 +6810,30 @@ tsk_squash_edges(tsk_edge_t *edges, size_t num_edges, size_t *num_output_edges)
     j = 0;
     l = 0;
     for (k = 1; k < num_edges; k++) {
-        assert(edges[k - 1].parent == edges[k].parent);
-        if (edges[k - 1].right != edges[k].left || edges[j].child != edges[k].child) {
-            e = edges[j];
-            e.right = edges[k - 1].right;
-            edges[l] = e;
+
+        if (edges[k-1].parent != edges[k].parent || edges[k-1].right != edges[k].left
+            || edges[j].child != edges[k].child) {
+
+            e.left = edges[j].left;
+            e.right = edges[k-1].right;
+            e.parent = edges[j].parent;
+            e.child = edges[j].child;
+
+            edges[j] = e;
+
             j = k;
             l++;
         }
     }
-    e = edges[j];
-    e.right = edges[k - 1].right;
```

Comment:
This blows up when there's zero edges. Need to add a test case to cover this in C.